From a50917945030556bfd7a39cb026d51fa9d286fd4 Mon Sep 17 00:00:00 2001
From: Jiang Kun <kun.jiang@intel.com>
Date: Mon, 16 Oct 2017 17:28:37 +0800
Subject: [PATCH 1443/2345] intel-ipu4: Trace the dual/multi camera for PnP
 KPIs

add "reserved" in vb2_v4l2_buffer. So the PnP KPI
can use it to trace the multi camera.

Signed-off-by: Jiang Kun <kun.jiang@intel.com>
---
 drivers/media/v4l2-core/videobuf2-v4l2.c | 2 +-
 include/media/videobuf2-v4l2.h           | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/v4l2-core/videobuf2-v4l2.c b/drivers/media/v4l2-core/videobuf2-v4l2.c
index da41f83..7e725fd 100644
--- a/drivers/media/v4l2-core/videobuf2-v4l2.c
+++ b/drivers/media/v4l2-core/videobuf2-v4l2.c
@@ -208,7 +208,7 @@ static void __fill_v4l2_buffer(struct vb2_buffer *vb, void *pb)
 	b->timecode = vbuf->timecode;
 	b->sequence = vbuf->sequence;
 	b->request = vbuf->request;
-	b->reserved = 0;
+	b->reserved = vbuf->reserved;
 
 	if (q->is_multiplanar) {
 		/*
diff --git a/include/media/videobuf2-v4l2.h b/include/media/videobuf2-v4l2.h
index 06f38523..d1995eb 100644
--- a/include/media/videobuf2-v4l2.h
+++ b/include/media/videobuf2-v4l2.h
@@ -43,6 +43,7 @@ struct vb2_v4l2_buffer {
 	struct v4l2_timecode	timecode;
 	__u32			sequence;
 	__u32			request;
+ 	__u32			reserved;
 };
 
 /*
-- 
2.7.4

